internal/cmdrpkgupdate/command.go

```diff
@@ -56,44 +57,53 @@ type runner struct {
 	Command *cobra.Command
 
 	revision string // Target package revision
+	discover bool   // If set, discover updates rather than do updates
+
+	// for --discover, there are multiple places where we need access to all package revisions, so
+	// we store it in the runner
+	prs []porchapi.PackageRevision
 }
 
 func (r *runner) preRunE(cmd *cobra.Command, args []string) error {
 	const op errors.Op = command + ".preRunE"
-	client, err := porch.CreateClient(r.cfg)
+	c, err := porch.CreateClient(r.cfg)
 	if err != nil {
 		return errors.E(op, err)
 	}
-	r.client = client
+	r.client = c
 
-	if len(args) < 1 {
-		return errors.E(op, fmt.Errorf("SOURCE_PACKAGE is a required positional argument"))
-	}
-	if len(args) > 1 {
-		return errors.E(op, fmt.Errorf("too many arguments; SOURCE_PACKAGE is the only accepted positional arguments"))
+	if !r.discover {
+		if len(args) < 1 {
+			return errors.E(op, fmt.Errorf("SOURCE_PACKAGE is a required positional argument"))
+		}
+		if len(args) > 1 {
+			return errors.E(op, fmt.Errorf("too many arguments; SOURCE_PACKAGE is the only accepted positional arguments"))
+		}
+		// TODO: This should use the latest available revision if one isn't specified.
+		if r.revision == "" {
+			return errors.E(op, fmt.Errorf("revision is required"))
+		}
 	}
 
-	// TODO: This should use the latest available revision if one isn't specified.
-	if r.revision == "" {
-		return errors.E(op, fmt.Errorf("revision is required"))
+	packageRevisionList := porchapi.PackageRevisionList{}
+	if err := r.client.List(r.ctx, &packageRevisionList, &client.ListOptions{}); err != nil {
+		return errors.E(op, err)
 	}
+	r.prs = packageRevisionList.Items
 
 	return nil
 }
 
 func (r *runner) runE(cmd *cobra.Command, args []string) error {
 	const op errors.Op = command + ".runE"
 
-	var pr porchapi.PackageRevision
-	err := r.client.Get(r.ctx, client.ObjectKey{
-		Namespace: *r.cfg.Namespace,
-		Name:      args[0],
-	}, &pr)
-	if err != nil {
-		return errors.E(op, err)
+	if r.discover {
+		return r.discoverUpdates(cmd, args)
 	}
 
-	cloneTask, found := r.findCloneTask(&pr)
+	pr := r.findPackageRevision(args[0])
```

Comment:
nit: Might be easier to read this if the code for updating a package was also pulled out into its own function like `discoverUpdates`.

Reply:
done!